### DISCUSSION TITLE
vuepress-plugin-redirect redirecting locales doesn't work in my project

### DISCUSSION THREAD

**Discussion by stydxm** [vuepress-theme-hope/vuepress-theme-hope#2936] [Q&A] [answered]
I don't know what's wrong, can anyone help me? (The feature is not enabled on the doc of the plugin, by the way, maybe someone can do this as a reference for users.)
part of my `package.json`
```json
"devDependencies": {
    "@vuepress/plugin-search": "2.0.0-beta.61",
    "@vuepress/plugin-shiki": "2.0.0-beta.61",
    "core-js-compat": "^3.28.0",
    "vuepress": "2.0.0-beta.61",
    "vuepress-theme-hope": "2.0.0-beta.197"
  }
```
`.vuepress/config.ts`
```ts
import { defineUserConfig } from "vuepress";
import { getDirname, path } from "@vuepress/utils";
import searchPlugin from "@vuepress/plugin-search";
import shikiPlugin from "@vuepress/plugin-shiki";
import { redirectPlugin } from "vuepress-plugin-redirect";
import theme from "./theme.js";

const __dirname = getDirname(import.meta.url);
export default defineUserConfig({
  lang: "en-US",
  title: "...",
  description: "...",
  locales: {
    "/": {
      lang: "en-US",
    },
    "/zh/": {
      lang: "zh-CN",
    },
  },
  theme,
  markdown: {
    importCode: {
      handleImportPath: (str) =>
        str.replace(/^@docs-root/, path.resolve(__dirname, "..")),
    },
  },
  plugins: [
    searchPlugin(),
    shikiPlugin({
      theme: "one-dark-pro",
    }),
    redirectPlugin({
      autoLocale: true,
      localeConfig: {
        "/": ["en-US", "en-UK", "en"],
        "/zh/": ["zh-CN", "zh-TW", "zh"],
      }
    }),
  ],
});

```

**Comment by Mister-Hope** [comment, marked as answer]
# READ THE Fxxxxxx DOCS PLZ 

![image](https://user-images.githubusercontent.com/33315834/226683782-550a8d4d-1247-4b0c-9fdf-d456c6f7ecbf.png)


https://plugin-redirect.vuejs.press/guide.html#redirecting-locales

A new feature with option `autoSwitchLocale: 'modal' | 'directly'` will be released in future.

**Reply by stydxm** [reply]
Sorry I didn't describe my problem clearly. I did try to move all docs except for `/zh` to `/en` and config everything as `/en`, but still does not work, so I leave a copy there and switched back. I apologize for my carelessness.

**Reply by Mister-Hope** [reply]
Track:

- https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/2933
- https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/2932

Everything should work at production, devServer redirection is not supported yet.

**Reply by stydxm** [reply]
I preview my project with dev server...Sorry I thought it's client side.
